Index: tools/roslaunch/src/roslaunch/core.py
===================================================================
--- tools/roslaunch/src/roslaunch/core.py	(revision 8072)
+++ tools/roslaunch/src/roslaunch/core.py	(working copy)
@@ -272,6 +272,7 @@
     ## start/restart master (i.e. restart an existing master)
     AUTO_RESTART = 2
     ZENMASTER = 'zenmaster'
+    ZENMASTER_SD = 'zenmaster_sd'
     BOTHERDER = 'botherder'    
 
     def __init__(self, type_=None, uri=None, auto=None):
Index: tools/roslaunch/src/roslaunch/nodeprocess.py
===================================================================
--- tools/roslaunch/src/roslaunch/nodeprocess.py	(revision 8072)
+++ tools/roslaunch/src/roslaunch/nodeprocess.py	(working copy)
@@ -85,8 +85,11 @@
     if type_ == Master.ZENMASTER:
         package = 'rospy'        
         args = [master, '--core', '-p', str(port)]
+    elif type_ == Master.ZENMASTER_SD:
+        package = 'zenmaster_sd'        
+        args = [master, '--core', '-p', str(port)]
     else:
-        raise RLException("unknown master typ_: %s"%type_)
+        raise RLException("unknown master type_: %s"%type_)
 
     _logger.info("process[master]: launching with args [%s]"%args)
     return LocalProcess(run_id, package, 'master', args, os.environ, log_output, None)
